src/nidigital/examples/burst_with_programmatic_configuration/nidigital_burst_with_programmatic_configuration.py

```diff
@@ -0,0 +1,184 @@
+            if voltage_config.termination_mode == 'High_Z':
+                session.channels[channels].termination_mode = nidigital.TerminationMode.HIGH_Z
+            elif voltage_config.termination_mode == 'Active_Load':
+                session.channels[channels].termination_mode = nidigital.TerminationMode.ACTIVE_LOAD
+                session.channels[channels].configure_active_load_levels(voltage_config.iol, voltage_config.ioh, voltage_config.vcom)
+            else:
+                session.channels[channels].termination_mode = nidigital.TerminationMode.VTERM
+
+        # Configure time set through driver API
+        if command == 'configure-time-set':
+            session.create_time_set(time_set_config.time_set_name)  # Must match time set name in pattern file
+            session.configure_time_set_period(time_set_config.time_set_name, time_set_config.period)
+            session.channels[channels].configure_time_set_drive_edges(time_set_config.time_set_name, convert_drive_format(time_set_config.drive_format),
+                                                                      time_set_config.drive_on, time_set_config.drive_data,
+                                                                      time_set_config.drive_return, time_set_config.drive_off)
+            session.channels[channels].configure_time_set_compare_edges_strobe(time_set_config.time_set_name, time_set_config.strobe_edge)
+
+        # Load the pattern file (.digipat) created using Digital Pattern Editor
+        pattern_filename = os.path.join(dir, 'Pattern.digipat')
+        session.load_pattern(pattern_filename)
+
+        # Burst pattern, blocks until the pattern is done bursting
+        session.burst_pattern(start_label='new_pattern')
+        print('Start bursting pattern')
+
+        # Disconnect all channels using programmable onboard switching
+        session.selected_function = nidigital.SelectedFunction.DISCONNECT
+    print('Done bursting pattern')
+
+
+def _main(argsv):
+    parser = argparse.ArgumentParser(description='Demonstrates how to create and configure an instrument session programmatically and to burst a pattern on the digital pattern instrument.', formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument('-n', '--resource-name', default='PXI1Slot2,PXI1Slot3', help='Resource name of a NI digital pattern instrument, ensure the resource name matches the instrument name in the pinmap file.')
+    parser.add_argument('-s', '--simulate', default='True', choices=['True', 'False'], help='Whether to run on simulated hardware or on real hardware')
+    subparser = parser.add_subparsers(dest='command', help='Sub-command help')
+
+    configure_voltage = subparser.add_parser('configure-voltage', help='Configure voltage levels and termination voltage of the digital pattern based on command line arguments')
```

Comment:
With the way it is, the user can choose to examine how configuring voltage works or how configuring time set works, but not both. It's like they can run only half the example at a time, which seems like weird restriction. Can you allow configuration of both simultaneously?

Reply:
It is possible just not with subcommands and the presentation is not as nice. I will just make the configure-voltage + configure-time-set a normal argument instead.

Reply:
Hmm... yeah, let's just keep it simple and not provide the subcommands at all. So the example will always use low-level API to configure both time set AND voltage. Only args will be the different time set and voltage values.

Then, you can also get rid off the Levels, Timing, and Specs files.

Reply:
Done